test-network-function/lifecycle/suite.go

```diff
@@ -82,89 +81,163 @@ var drainTimeout = time.Duration(drainTimeoutMinutes) * time.Minute
-func testDeployments(namespace string) {
+func testShutdown(configData *common.ConfigurationData) {
+	ginkgo.When("Testing PUTs are configured with pre-stop lifecycle", func() {
+		ginkgo.It("should have pre-stop configured", func() {
+			for _, cut := range configData.ContainersUnderTest {
+				podName := cut.Oc.GetPodName()
+				podNamespace := cut.Oc.GetPodNamespace()
+				ginkgo.By(fmt.Sprintf("should have pre-stop configured %s/%s", podNamespace, podName))
+				defer results.RecordResult(identifiers.TestShudtownIdentifier)
+				shutdownTest(podNamespace, podName)
+			}
+		})
+	})
+}
+
+func shutdownTest(podNamespace, podName string) {
+	context := common.GetContext()
+	values := make(map[string]interface{})
+	values["POD_NAMESPACE"] = podNamespace
+	values["POD_NAME"] = podName
+	values["GO_TEMPLATE_PATH"] = relativeShutdownTestDirectoryPath
+	test, handlers, result, err := generic.NewGenericFromMap(relativeShutdownTestPath, common.RelativeSchemaPath, values)
+	gomega.Expect(err).To(gomega.BeNil())
+	gomega.Expect(result).ToNot(gomega.BeNil())
+	gomega.Expect(result.Valid()).To(gomega.BeTrue())
+	gomega.Expect(handlers).ToNot(gomega.BeNil())
+	gomega.Expect(handlers).ToNot(gomega.BeNil())
+	gomega.Expect(test).ToNot(gomega.BeNil())
+	tester, err := tnf.NewTest(context.GetExpecter(), *test, handlers, context.GetErrorChannel())
+	gomega.Expect(err).To(gomega.BeNil())
+	gomega.Expect(tester).ToNot(gomega.BeNil())
+
+	testResult, err := tester.Run()
+	gomega.Expect(err).To(gomega.BeNil())
+	gomega.Expect(testResult).To(gomega.Equal(tnf.SUCCESS))
+	gomega.Expect(testResult).To(gomega.Equal(tnf.SUCCESS))
+}
+
+func testDeployments(configData *common.ConfigurationData) {
 	var deployments dp.DeploymentMap
 	var notReadyDeployments []string
+	var nodesSorted []node // A slice version of nodes sorted by number of deployments descending
 	ginkgo.When("Testing deployments in namespace", func() {
 		ginkgo.It("Should return list of deployments", func() {
-			deployments, notReadyDeployments = getDeployments(namespace)
-			if len(deployments) == 0 {
-				return
+			configData.SetNeedsRefresh()
+			for _, cut := range configData.ContainersUnderTest {
+				namespace := cut.Oc.GetPodNamespace()
+				ginkgo.By(fmt.Sprintf("test deployment in namespace %s", namespace))
+				deployments, notReadyDeployments = getDeployments(namespace)
+				if len(deployments) == 0 {
+					return
+				}
+				// We require that all deployments have the desired number of replicas and are all up to date
+				if len(notReadyDeployments) == 0 {
+					ginkgo.Skip("Can not test when deployments are not ready")
+				}
+				gomega.Expect(notReadyDeployments).To(gomega.BeEmpty())
+				ginkgo.By("Should return map of nodes to deployments")
+				nodesSorted = getDeploymentsNodes(namespace)
+				if !common.NonIntrusive() {
+					ginkgo.By("should create new replicas when node is drained")
+					defer results.RecordResult(identifiers.TestPodRecreationIdentifier)
```

Comment:
This identifier was removed from the catalog map. Add it back if we are including the test now.

Reply:
fixed.